### PR TITLE
FE-768 | Update tests for CreateAccessProvider

### PR DESCRIPTION
**Pull request by fauna-ryanharris**
### Notes
[Jira Ticket](https://faunadb.atlassian.net/browse/FE-768)

**NOTE:** #339 needs to be merged first in order for these tests to pass CI

This PR updates the test coverage for `CreateAccessProvider` to reflect the new functionality. Specifically, we now have a `membership` field whose value is an array of Roles.

### How to test
Using the `nightly` version of our Core image locally, you can run `npm run test`

### CODE REVIEW

File: test/query.test.js [hunk 1]
```diff
@@ -861,25 +861,63 @@ describe('query', () => {
...
-    expect(provider.jwks_uri).toEqual(fullUri)
+    // Successful instantiation with required fields
+    try {
+      const provider = await adminClient.query(
+        query.CreateAccessProvider({
+          name: providerName,
+          issuer: issuerName,
+          jwks_uri: fullUri,
+          membership: [
+            myNewRole,
```
Comment (marrony) [inline test/query.test.js:905]: You are passing the role document here, you should pass the reference

Comment (fauna-ryanharris) [reply test/query.test.js:905]: @marrony So it would be `Role(roleOneName)` instead?

File: test/query.test.js [hunk 2]
```diff
@@ -861,25 +861,63 @@ describe('query', () => {
...
+
+    // Successful instantiation with required fields
+    try {
+      const provider = await adminClient.query(
+        query.CreateAccessProvider({
+          name: providerName,
+          issuer: issuerName,
+          jwks_uri: fullUri,
+          membership: [
+            query.Role(`${roleOneName}`),
```
Comment (fauna-ryanharris) [inline test/query.test.js:905]: @marrony Like this?

Comment (marrony) [reply test/query.test.js:905]: yes.

I don't get why you are using string interpolation here, isn't `roleOneName` a string already? so it would be the same `query.Role(roleOneName)`

Comment (fauna-ryanharris) [inline test/query.test.js:905]:
```suggestion
            query.Role(roleOneName),
```

File: test/query.test.js [hunk 3]
```diff
@@ -1569,14 +1607,6 @@ describe('query', () => {
     return Promise.all([p1, p2, p3, p4])
   })
 
-  test('contains', () => {
```
Comment (marrony) [inline test/query.test.js:1572]: I think this should belong to a separate PR

File: src/query.js [hunk 4]
```diff
@@ -1032,8 +1032,7 @@ function CreateRole(params) {
  *     - name: A valid schema name
  *     - issuer: A unique string
  *     - jwks_uri: A valid HTTPS URI
- *     - allowed_roles: An optional list of Role refs
- *     - allowed_collections: An optional list of user-defined Collection refs
+ *     - membership: An array of roles
```
Comment (marrony) [inline src/query.js:1035]: Array of roles doesn't convey the correct purpose of this field. It should say it accepts an object with a pair of role/predicate and that the predicate returns true or false.

File: test/query.test.js [hunk 5]
```diff
@@ -861,25 +861,63 @@ describe('query', () => {
...
+    )
+
+    // Successful instantiation with required fields
+    try {
+      const provider = await adminClient.query(
+        query.CreateAccessProvider({
+          name: providerName,
+          issuer: issuerName,
+          jwks_uri: fullUri,
+          membership: [
```
Comment (nathaniel-may) [inline test/query.test.js:904]: I like that you used both kinds of membership values here 👍

File: test/query.test.js [hunk 6]
```diff
@@ -861,25 +861,63 @@ describe('query', () => {
...
+          ],
+        })
+      )
+
+      expect(provider.name).toEqual(providerName)
+      expect(provider.issuer).toEqual(issuerName)
+      expect(provider.jwks_uri).toEqual(fullUri)
+      expect(provider.membership).toBeInstanceOf(Array)
+    } catch (error) {
+      expect(error).toBeInstanceOf(errors.BadRequest)
```
Comment (nathaniel-may) [inline test/query.test.js:919]: Does this mean that if the try block fails, the test will still pass if the error is a BadRequest error? I imagine we'd want to require that the try block is successful.

Comment (marrony) [reply test/query.test.js:919]: yeah, try/catch on tests might not be a good practice, you need to be sure your code will always succeed or fail and do the correct assertions.

Comment (fauna-ryanharris) [reply test/query.test.js:919]: @marrony @nathaniel-may I'll split these tests in two so we can make assertions re: success and failure

**Review by nathaniel-may — APPROVED** [review]
LGTM
